src/tools/blast/components/results/HSPDetailPanel.tsx

```diff
@@ -0,0 +1,383 @@
+
+        const blockSegments = seqSegments.map(([start, end]) => ({
+          start: start + offset,
+          end: end + offset,
+          // franklin $colour-sapphire-blue
+          color: '#014371',
+          opacity,
+        }));
+
+        const lineSegments = [
+          {
+            start: 1,
+            end: offset,
+            shape: 'line',
+            color: '#014371',
+            opacity,
+          },
+          {
+            start: blockSegments[blockSegments.length - 1].end + 1,
+            end: hitLength,
+            shape: 'line',
+            color: '#014371',
+            opacity,
+          },
+        ];
+
+        // eslint-disable-next-line no-param-reassign
+        node.data = [...lineSegments, ...blockSegments];
+      }
+    },
+    [hsp_hit_from, hsp_hit_to, hsp_identity, hitLength]
+  );
+
+  const setMSAAttributes = useCallback(
+    (node): void => {
+      if (!node) {
+        return;
+      }
+      node.data = [
+        {
+          name: 'Query',
+          sequence: hsp_qseq,
+        },
+        {
+          name: 'Match',
+          sequence: hsp_hseq,
+        },
+      ];
+
+      node.onActiveTrackChange = (trackId: string) => {
+        setActiveTrack(trackId);
+        console.log('on active track change:', trackId);
+      };
+    },
+    [hsp_qseq, hsp_hseq]
+  );
+
+  const { loading, data, status, error } = useDataApi<UniProtkbAccessionsAPI>(
+    apiUrls.entries([hitAccession])
+  );
+
+  const features = data?.results?.[0]?.features;
+  const recommendedName =
+    data?.results?.[0]?.proteinDescription?.recommendedName?.fullName.value;
+  const organism = data?.results?.[0]?.organism?.scientificName;
+
+  const title = [hitAccession, recommendedName, organism]
+    .filter(Boolean)
+    .join(' · ');
+
+  const annotationChoices = uniq(features?.map(({ type }) => type));
+  if (!annotation && !!annotationChoices?.length) {
+    setAnnotation(annotationChoices[0]);
+  }
+
+  const setFeatureTrackData = useCallback(
+    (node): void => {
+      if (node && features && annotation) {
+        const processedFeatures = processFeaturesData(
+          features.filter(({ type }) => type === annotation)
+        );
+        node.data = processedFeatures;
+      }
+    },
+    [features, annotation]
+  );
+
+  type EventDetail = {
+    displaystart: string;
+    displayend: string;
+  };
+
+  const findHighlighPositions = ({ displaystart, displayend }: EventDetail) => {
+    setHighlighPosition(`${displaystart}:${displayend}`);
+  };
+
+  const managerRef = useCallback((node): void => {
+    if (node) {
+      node.addEventListener('change', ({ detail }: { detail: EventDetail }) =>
+        findHighlighPositions(detail)
+      );
+    }
+  }, []);
+
+  if (error) {
+    return <ErrorHandler status={status} />;
+  }
+
+  if (loading) {
+    return <Loader />;
+  }
+
+  return (
+    <SlidingPanel position="bottom">
+      <>
+        {/* TODO move to stylesheet */}
```

Comment:
Indeed

Reply:
Moved these styles to its own file for HSP details panel, but let me know if they should have gone to `SlidingPanel` styles.